systems/controllers/QPCommon.cpp

```diff
@@ -100,6 +100,24 @@ std::shared_ptr<drake::lcmt_qp_controller_input> encodeQPInputLCM(const mxArray
     }
   }
 
+  const mxArray* body_wrench_data = myGetProperty(qp_input, "body_wrench_data");
+  const int num_external_wrenches = mxGetN(body_wrench_data);
+  msg->num_external_wrenches = num_external_wrenches;
+  msg->body_wrench_data.resize(num_external_wrenches);
+  const int wrench_size = 6;
+  if (num_external_wrenches > 0) {
+    if (mxGetM(body_wrench_data) != 1) {
+      mexErrMsgTxt("body wrench data should be a 1xN struct array");
+    }
+    for (int i = 0; i < num_external_wrenches; i++) {
+      msg->body_wrench_data[i].timestamp = msg->timestamp;
+      msg->body_wrench_data[i].body_id = (int32_t) mxGetScalar(myGetField(body_wrench_data, i, "body_id"));
```

Comment:
I disagree: I've been making sure that the LCM and matlab data structures contain exactly the same data, rather than pre-emptively subtracting one from all the IDs that get passed over LCM.